src/main/java/org/wso2/carbon/connector/KafkaProduceConnector.java

```diff
@@ -584,6 +588,137 @@ private Object handleComplexAvroTypes(Object jsonString, Schema schema) {
         }
     }
 
+    private Object handleTypeInt(String input, Schema schema) {
+        if (LogicalTypes.date().equals(schema.getLogicalType())) {
+            try {
+                Date date = DateUtils.parseDate(input, "yyyy-MM-dd");
+                return Utils.convertFromDate(date);
+            } catch (ParseException e) {
+                throw new SerializationException("Error serializing Avro message of type int with logicalType "
+                        + "date for input: " + input + ". The input must be an integer representing "
```

Comment:
Saying `input must be an integer` maybe confusing since we expect a value in the `yyyy-MM-dd` format

Reply:
Fixed